docs/4.4/api-reference.md

```diff
@@ -54,14 +54,131 @@ certificate lifetime.
 
 # gRPC APIs
 
+## Client QuickStart
+
+```go
+{!examples/go-client/client.go!}
+```
+
 ## Audit Events API
 Coming Soon
 
 ## Certificate Generation API
 Coming Soon
 
+## Roles APIs
+
+[Roles](http://localhost:6600/enterprise/ssh-rbac/#roles) are used to define what resources and actions a user is allowed/denied access to.
+
+#### Create a new role
+
+```go
+import github.com/gravitational/teleport/lib/services
+
+role, _ := services.NewRole(roleName, services.RoleSpecV3{
+	Allow: services.RoleConditions{
+    Logins: []string{login},
+	},
+})
+
+client.UpsertRole(ctx, role)
+```
+
+#### retrieve roles
+
+```go
+roles, err := client.GetRoles()
+if err != nil {
+  return err
+}
+```
+
+#### update role
+
+```go
+// update a role to deny access to nodes labeled as production
+// you can update any attributes a role has with its setter methods
+role, err = client.GetRole("dev")
+if err != nil {
+  log.Fatal(err)
+}
+  
+role.SetNodeLabels(services.Deny, services.Labels{"environment": "production"})
+if err = client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
+
+#### delete role
+
+```go
+if err = client.DeleteRole(ctx, "dev"); err != nil {
+  return err
+}
+```
+
 ## Tokens API
-Coming Soon
+
+[Tokens](http://localhost:6600/admin-guide/#adding-nodes-to-the-cluster) are used to add nodes to a cluster.
+
+#### Create a new token
+
+```go
+tokenName := "mytoken"
```

Comment:
better use strong crypto random hex token in the example to set a good precedent